utils/ixdtf/src/parsers/datetime.rs

```diff
@@ -78,13 +79,24 @@ pub(crate) fn parse_annotated_date_time<'a>(
 }
 
 /// Parses a `DateTime` record.
-fn parse_date_time(cursor: &mut Cursor) -> ParserResult<DateTimeRecord> {
-    let date = parse_date(cursor)?;
+fn parse_date_time(cursor: &mut Cursor, options: IxdtfOptions) -> ParserResult<DateTimeRecord> {
+    if options == IxdtfOptions::Time {
+        let start = cursor.pos;
+        let result = parse_ambiguous_time_record(cursor)?;
+        if let Some(success) = result {
```

Comment:
Suggestion/Issue: I don't think I like the graceful recovery here. If you pass `IxdtfOptions::MonthDay`, and the first field is not a month, then the parse should fail, and if desired, the caller can try again with a different `IxdtfOptions`. It also makes it harder to reason about the code since these inner functions like `parse_ambiguous_time_record` might do funny things with `cursor`.